Nos/Views/Moderation/UserFlagView.swift

```diff
@@ -70,44 +81,59 @@ struct UserFlagView: View {
-                        title: String(localized: .localizable.flagSendTitle),
-                        subtitle: nil
-                    )
-                    .transition(.move(edge: .leading).combined(with: .opacity))
-                }
-
-                if selectedSendOption != nil {
-                    FlagOptionPicker(
-                        previousSelection: .constant(nil),
-                        currentSelection: $selectedVisibilityOption,
-                        options: FlagOption.flagUserVisibilityOptions,
-                        title: String(localized: .localizable.flagUserMuteCategoryTitle),
-                        subtitle: nil
-                    )
-                    .transition(.move(edge: .leading).combined(with: .opacity))
+        GeometryReader { geometry in
+            ScrollViewReader { proxy in
+                ScrollView(showsIndicators: false) {
+                    VStack(spacing: 30) {
+                        FlagOptionPicker(
+                            previousSelection: .constant(nil),
+                            currentSelection: $selectedFlagOption,
+                            options: flagCategories,
+                            title: String(localized: .localizable.flagUserCategoryTitle),
+                            subtitle: String(localized: .localizable.flagUserCategoryDescription)
+                        )
+                        
+                        if selectedFlagOption != nil {
+                            FlagOptionPicker(
+                                previousSelection: $selectedFlagOption,
+                                currentSelection: $selectedSendOption,
+                                options: FlagOption.flagUserSendOptions,
+                                title: String(localized: .localizable.flagSendTitle),
+                                subtitle: nil
+                            )
+                            .id(sendSectionID)
+                            .offset(x: animateSendSection ? 0 : -geometry.size.width)
+                        }
+                        
+                        if selectedSendOption != nil {
+                            FlagOptionPicker(
+                                previousSelection: .constant(nil),
+                                currentSelection: $selectedVisibilityOption,
+                                options: FlagOption.flagUserVisibilityOptions,
+                                title: String(localized: .localizable.flagUserMuteCategoryTitle),
+                                subtitle: nil
+                            )
+                            .id(visibilitySectionID)
+                            .offset(x: animateVisibilitySection ? 0 : -geometry.size.width)
+                        }
+                    }
+                    .padding()
+                    .onChange(of: selectedFlagOption) {
+                        /// Animates the sliding effect and scrolls to the bottom of the specified section.
```

Comment:
I think we could do without these comments (here and the one below, on line 131), but I'll leave it up to you. If you like them, they're fine with me.

Reply:
Thanks @joshuatbrown , I just wanted to be sure anyone reading it understands the type of animation being added 😃